try to run make deps if it exists

whyrusleeping opened the pull request:
This should allow us to build 0.4.0 (master) correctly now that we use gx vendoring.

whyrusleeping commented:
@noffle wanna take a look at this one?

dignifiedquire commented:
LGTM

Kubuxu commented on build-go.sh:
```diff
@@ -201,6 +201,14 @@ function checkoutVersion() {
+
+	reporoot=$(cd "$repopath" && git rev-parse --show-toplevel)
+
+	(cd "$reporoot" && make deps)
```
It might be over-engineering but calling:
```bash
(cd "$reporoot" && make -n deps >/dev/null 2>&1 && make deps)
```
seems much cleaner to me.
`-n` flag makes dry run so we can check if given rule exists.

whyrusleeping replied:
ooooh, i was trying to find that flag in the manpages, thanks!

noffle commented:
I'm seeing the following on the `gx install` step (pre-build) for `v0.4.0-rc2`:
```
installation of gogo-protobuf complete!
  - running post install for gogo-protobuf: /home/stephen/Projects/Forks/distributions/dists/go-ipfs/gopath/src/gx/ipfs/QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV
rewrite error:  /home/stephen/Projects/Forks/distributions/dists/go-ipfs/gopath/src/gx/ipfs/QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV/gogo-protobuf/test/packed/packed.pb.go:1907:23: string literal not terminated
ERROR: rewrite failed:  /home/stephen/Projects/Forks/distributions/dists/go-ipfs/gopath/src/gx/ipfs/QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV/gogo-protobuf/test/packed/packed.pb.go:1907:23: string literal not terminated
ERROR: install deps: post-install hook failed: exit status 1
Makefile:22: recipe for target 'deps' failed
make[2]: *** [deps] Error 1
make[2]: Leaving directory '/home/stephen/Projects/Forks/distributions/dists/go-ipfs/gopath/src/github.com/ipfs/go-ipfs'
```

noffle commented:
Could we redirect the verbose `gx` noise to a log somewhere? Maybe a `fetch-log` or `prebuild-log`? Something with [per-arch/os names](https://github.com/ipfs/distributions/pull/57) would be preferable!

whyrusleeping commented:
@noffle that looks kinda like that file in question didnt get downloaded properly... my guess is that a previous download was interupted and gx assumed that the previous download completed successfully.

noffle commented:
That must've been it. I am, however, still seeing `iana` errors for everything but linux-(all arches):

```
../gopath/src/gx/ipfs/QmZy2y8t9zQH2a1b8q2ZSLKp17ATuJoCNxxyMFG5qFExpt/go-net/ipv4/sockopt_asmreq_windows.go:13:2: cannot find package "golang.org/x/net/internal/iana" in any of:
	/usr/lib/go/src/golang.org/x/net/internal/iana (from $GOROOT)
	/home/stephen/Projects/Forks/distributions/dists/go-ipfs/gopath/src/golang.org/x/net/internal/iana (from $GOPATH)
```

whyrusleeping commented:
@dignifiedquire i have the script set up so that it outputs a patched object hash after building new versions of a distribution. But i'm not sure the best way to wire that up to the build as a whole.

lgierth commented:
I gave it a try and noticed a few small things:

- it seems to be hardwired to a local daemon
- when fetch /ipns/dist.ipfs.io/go-ipfs, after ~55MB I start getting "too many files open" from the daemon, on Linux (!) with ulimit being unlimited (!) :P
- when I kill the daemon during fetch, `make all` falls back to building everything locally
- if `make all` has started to build locally once, subsequent attempts will not try the daemon, but instead directly build everything. I did `git clean -fdx` to recover from this.
- `make deps` doesn't seem to include the site deps

I'm trying to prefetch all of it via `ipfs refs -r /ipns/dist.ipfs.io` now.

whyrusleeping commented:
@lgierth i've addressed some of those concerns. Made it a little easier to stop and restart without scewing things up (as much). and failed partial downloads of the existing distrbutions page won't bork the next run.
